.github/workflows/build-test-distribute.yaml

```diff
@@ -170,3 +173,153 @@ jobs:
           trap on_exit EXIT
           make docker/push
           make docker/manifest
+  gen_e2e_matrix:
+    runs-on: ubuntu-latest
+    needs: ["build"]
+    outputs:
+      slow: ${{ steps.generate-matrix.outputs.slow }}
+      main: ${{ steps.generate-matrix.outputs.main }}
+      delta_kds: ${{ steps.generate-matrix.outputs.delta_kds }}
+      cni: ${{ steps.generate-matrix.outputs.cni }}
+    steps:
+      - name: Install jq
+        run: |
+          sudo apt-get install -y jq
+      - id: generate-matrix
+        name: Generate matrix
+        run: |
+          SKIP_CI_LABEL="${{ contains(github.event.pull_request.labels.*.name, 'ci/skip-test') || contains(github.event.pull_request.labels.*.name, 'ci/skip-e2e-test')  }}"
+          if [[ "$SKIP_CI_LABEL" == "true" ]]; then
+            echo "Skipping e2e jobs because the pull request contained a 'ci/skip-*' label."
+            exit 0
+          fi
```

Comment:
I think this would be nicer as an `if` at the github job level like we do for the rest (also speeds things up)

Reply:
Sure thing, will do